tests/scripts/findkernelversion.sh

```diff
@@ -17,8 +17,28 @@ chmod a+x bin/regctl
 export PATH=$(pwd)/bin:${PATH}
 
 # calculate kernel version of latest image
-regctl image get-file ghcr.io/nvidia/driver:base-${BASE_TARGET}-${KERNEL_FLAVOR}-${DRIVER_BRANCH} /var/kernel_version.txt ./kernel_version.txt
-export $(grep -oP 'KERNEL_VERSION=[^ ]+' ./kernel_version.txt)
+regctl image get-file ghcr.io/nvidia/driver:base-${BASE_TARGET}-${KERNEL_FLAVOR}-${DRIVER_BRANCH} /var/kernel_version.txt ./kernel_version.txt 2>/dev/null || true
+if [[ -f ./kernel_version.txt && -s ./kernel_version.txt ]]; then
+    # File exists and is not empty
+    export $(grep -oP 'KERNEL_VERSION=[^ ]+' ./kernel_version.txt)
+    rm -f kernel_version.txt
+else
+    # Define variables for artifact pattern
+    prefix="kernel-version-${DRIVER_BRANCH}-${LTS_KERNEL}"
+    suffix="${kernel_flavor}-${DIST}"
+    artifacts=$(gh api -X GET /repos/${GITHUB_REPOSITORY}/actions/runs/${GITHUB_RUN_ID}/artifacts --jq '.artifacts[].name')
+    # Use a loop or a pattern to find the matching artifact dynamically
+    for artifact in $artifacts; do
+        if [[ $artifact == $prefix*-$suffix ]]; then
+            gh run download --name "$artifact" --dir ./
+            tar -xf $artifact.tar 
+            rm -f $artifact.tar
+            export $(grep -oP 'KERNEL_VERSION=[^ ]+' ./kernel_version.txt)
+            rm -f kernel_version.txt
+            break
+        fi
+    done
+fi
```

Comment:
Question -- why was this added?

Reply:
added for first release of ubuntu24.04 or other distro